pkg/trace/api/normalizer_test.go

```diff
@@ -33,277 +36,326 @@ func testSpan() *pb.Span {
 	assert.Equal(t, before, s.Metrics)
+	assertNoNormalizationIssues(t, ts)
 }
 
 func TestNormalizeMetaPassThru(t *testing.T) {
-	s := testSpan()
+	ts, s := &info.TagStats{}, newTestSpan()
 	before := s.Meta
-	normalize(s)
+	assert.NoError(t, normalize(ts, s))
 	assert.Equal(t, before, s.Meta)
+	assertNoNormalizationIssues(t, ts)
 }
 
 func TestNormalizeParentIDPassThru(t *testing.T) {
-	s := testSpan()
+	ts, s := &info.TagStats{}, newTestSpan()
 	before := s.ParentID
-	normalize(s)
+	assert.NoError(t, normalize(ts, s))
 	assert.Equal(t, before, s.ParentID)
+	assertNoNormalizationIssues(t, ts)
 }
 
 func TestNormalizeTypePassThru(t *testing.T) {
-	s := testSpan()
+	ts, s := &info.TagStats{}, newTestSpan()
 	before := s.Type
-	normalize(s)
+	assert.NoError(t, normalize(ts, s))
 	assert.Equal(t, before, s.Type)
+	assertNoNormalizationIssues(t, ts)
 }
 
 func TestNormalizeTypeTooLong(t *testing.T) {
-	s := testSpan()
+	ts, s := &info.TagStats{}, newTestSpan()
 	s.Type = strings.Repeat("sql", 1000)
-	normalize(s)
-	assert.Error(t, normalize(s))
+	assert.NoError(t, normalize(ts, s))
+	assertNormalizationIssue(t, ts, "type_truncate")
 }
 
 func TestNormalizeServiceTag(t *testing.T) {
-	s := testSpan()
+	ts, s := &info.TagStats{}, newTestSpan()
 	s.Service = "retargeting(api-Staging "
-	normalize(s)
+	assert.NoError(t, normalize(ts, s))
 	assert.Equal(t, "retargeting_api-staging", s.Service)
+	assertNoNormalizationIssues(t, ts)
 }
 
 func TestNormalizeEnv(t *testing.T) {
-	s := testSpan()
+	ts, s := &info.TagStats{}, newTestSpan()
 	s.Meta["env"] = "DEVELOPMENT"
-	normalize(s)
+	assert.NoError(t, normalize(ts, s))
 	assert.Equal(t, "development", s.Meta["env"])
+	assertNoNormalizationIssues(t, ts)
 }
 
 func TestSpecialZipkinRootSpan(t *testing.T) {
-	s := testSpan()
+	ts, s := &info.TagStats{}, newTestSpan()
 	s.ParentID = 42
 	s.TraceID = 42
 	s.SpanID = 42
 	beforeTraceID := s.TraceID
 	beforeSpanID := s.SpanID
-	normalize(s)
+	assert.NoError(t, normalize(ts, s))
 	assert.Equal(t, uint64(0), s.ParentID)
 	assert.Equal(t, beforeTraceID, s.TraceID)
 	assert.Equal(t, beforeSpanID, s.SpanID)
+	assertNoNormalizationIssues(t, ts)
 }
 
 func TestNormalizeTraceEmpty(t *testing.T) {
-	trace := pb.Trace{}
-
-	err := normalizeTrace(trace)
+	ts, trace := &info.TagStats{}, pb.Trace{}
+	err := normalizeTrace(ts, trace)
 	assert.Error(t, err)
+	assertNormalizationIssue(t, ts, "empty_trace")
 }
 
 func TestNormalizeTraceTraceIdMismatch(t *testing.T) {
-	span1 := testSpan()
+	ts, span1, span2 := &info.TagStats{}, newTestSpan(), newTestSpan()
 	span1.TraceID = 1
-
-	span2 := testSpan()
 	span2.TraceID = 2
-
 	trace := pb.Trace{span1, span2}
-
-	err := normalizeTrace(trace)
+	err := normalizeTrace(ts, trace)
 	assert.Error(t, err)
+	assertNormalizationIssue(t, ts, "foreign_span")
 }
 
 func TestNormalizeTraceInvalidSpan(t *testing.T) {
-	span1 := testSpan()
-
-	span2 := testSpan()
+	ts, span1, span2 := &info.TagStats{}, newTestSpan(), newTestSpan()
```

Comment:
Same here and everywhere else please 🙏. Declaring variables like this is a bit confusing. We're not really doing it anywhere. As far as I've learned, the only idiomatic and valid reasons to do this, in general, are:
* for loops with multiple variables
* if statements
```go
if n1, n2 := len(a), len(b); n1 != n2 {
    // ...
}
```
* swaps:
```go
a, b = b, a
```
I'm sure some rare case can be found where it makes sense within declarations too, but I haven't found it and this doesn't look like one of them.

Reply:
Ok, updated. I figured it was simpler to do it all one one line, but fine like this as well.